language/move-prover/interpreter/src/concrete/player.rs

```diff
@@ -0,0 +1,1357 @@
+            .string(target.get_local_name(i))
+            .to_string();
+        let ty = convert_model_local_type(env, target.get_local_type(i), &ctxt.ty_args);
+        let slot = LocalSlot::new_tmp(name, ty);
+        local_slots.push(slot);
+    }
+
+    // execute the bytecode stepwise
+    let instructions = target.get_bytecode();
+    let mut local_state = LocalState::new(local_slots);
+    while !local_state.is_terminated() {
+        let pc = local_state.get_pc() as usize;
+        let bytecode = instructions.get(pc).unwrap();
+        exec_bytecode(ctxt, bytecode, &mut local_state, global_state)?;
+    }
+    Ok(local_state)
+}
+
+fn exec_bytecode(
+    ctxt: &FunctionContext,
+    bytecode: &Bytecode,
+    local_state: &mut LocalState,
+    global_state: &mut GlobalState,
+) -> PartialVMResult<()> {
+    match bytecode {
+        Bytecode::Assign(_, dst, src, kind) => handle_assign(*dst, *src, kind, local_state),
+        Bytecode::Load(_, dst, constant) => handle_load(*dst, constant, local_state),
+        Bytecode::Call(_, dsts, op, srcs, on_abort) => handle_operation(
+            ctxt,
+            dsts,
+            op,
+            srcs,
+            on_abort.as_ref(),
+            local_state,
+            global_state,
+        )?,
+        Bytecode::Label(_, label) => {
+            if cfg!(debug_assertions) {
+                ctxt.code_offset_by_label(*label);
+            }
+        }
+        Bytecode::Jump(_, label) => {
+            local_state.set_pc(ctxt.code_offset_by_label(*label));
+        }
+        Bytecode::Branch(_, then_label, else_label, cond) => {
+            handle_conditional_branch(ctxt, *cond, *then_label, *else_label, local_state)
+        }
+        Bytecode::Abort(_, index) => handle_abort(*index, local_state),
+        Bytecode::Ret(_, rets) => handle_return(ctxt, rets, local_state),
+        // global memory and expressions (TODO: not supported yet)
+        Bytecode::SaveMem(..) | Bytecode::Prop(..) => {}
+        // deprecated
+        Bytecode::Nop(_) | Bytecode::SpecBlock(..) | Bytecode::SaveSpecVar(..) => {}
+    }
+    local_state.ready_pc_for_next_instruction();
+    Ok(())
+}
+
+//**************************************************************************************************
+// Per-bytecode handling
+//**************************************************************************************************
+
+fn handle_assign(dst: TempIndex, src: TempIndex, kind: &AssignKind, local_state: &mut LocalState) {
+    // check types
+    if cfg!(debug_assertions) {
+        assert!(local_state
+            .get_type(dst)
+            .is_compatible_for_assign(local_state.get_type(src)));
+    }
+
+    // execute
+    let slot = match kind {
+        AssignKind::Move => local_state.del_value(src),
+        // TODO (mengxu): what exactly is the semantic of Store here? Why not just use Copy?
+        AssignKind::Copy | AssignKind::Store => local_state.get_value(src).deref(),
```

Comment:
I think Store is a leftover from old days and really doesn't differ from Copy (or Move!). I believe its generated if something is popped from the stack and stored in a temp. 

Whether its Move or Copy (if this is relevant here) is best seen in the borrow analysis in bytecode.

Reply:
I see `Store` is treated like `Move` everywhere in the pipeline (and in boogie translator as well), but when executing it, I found that `dst := Store(src)` does not clear the local slot at `src`. So I treated it as a `Copy` here to keep things going. I'll come back with a better handling and remove the TODO.